koji_containerbuild/plugins/builder_containerbuild.py

```diff
@@ -432,34 +425,40 @@ def createContainer(self, src=None, target_info=None, arch=None, arches=None,
         scm.assert_allowed(self.options.allowed_scms)
         git_uri = scm.get_git_uri()
         component = scm.get_component()
+        arch = None
+
+        if not arches:
+            raise ContainerError("arches aren't specified")
 
         create_build_args = {
             'git_uri': git_uri,
             'git_ref': scm.revision,
             'user': owner_info['name'],
             'component': component,
             'target': target_info['name'],
-            'architecture': arch,
             'yum_repourls': yum_repourls,
             'scratch': scratch,
             'koji_task_id': self.id,
+            'architecture': arch
         }
-        if arches:
-            create_build_args['platforms'] = arches
-            create_method = self.osbs().create_orchestrator_build
-        elif arch:
-            create_build_args['platform'] = arch
-            create_method = self.osbs().create_build
-        else:
-            raise ContainerError("Neither arch nor arches specified")
-
         if branch:
             create_build_args['git_branch'] = branch
         if push_url:
             create_build_args['git_push_url'] = push_url
 
-        self.logger.debug("Starting %s with params: '%s'", create_method, create_build_args)
-        build_response = create_method(**create_build_args)
+        try:
+            create_method = self.osbs().create_orchestrator_build
+            self.logger.debug("Starting %s with params: '%s, platforms:%s'",
+                              create_method, create_build_args, arches)
+            build_response = create_method(platforms=arches, **create_build_args)
```

Comment:
Agree with Tim and Lui on this - there is no point to use `create_method` if it would be used just once.

This might be a good idea to print its name in some error message though